decision_service/examples/rl_sim_cpp/rl_sim.cc

```diff
@@ -0,0 +1,127 @@
+#include <iostream>
+#include <fstream>
+#include <boost/uuid/random_generator.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include "rl_sim.h"
+#include "ranking_response.h"
+#include "person.h"
+#include "live_model.h"
+#include "rl_sim_cpp.h"
+#include "config_utility.h"
+
+namespace r = reinforcement_learning;
+namespace u = reinforcement_learning::utility;
+namespace c = reinforcement_learning::utility::config;
+
+rl_sim::rl_sim(boost::program_options::variables_map vm):_options(std::move(vm)) {}
+std::string rl_sim::create_context_json(const std::string& cntxt, const std::string& action) {
+  std::ostringstream oss;
+  oss << "{ " << cntxt << ", " << action << " }";
+  return oss.str();
+}
+
+std::string rl_sim::create_uuid() {
+  return boost::uuids::to_string(boost::uuids::random_generator()( ));
+}
+
+int rl_sim::loop() {
+  if ( !init() ) return r::error_code::success;
+  
+  auto round = 0;
+  r::ranking_response response;
+  while(true) {
+    auto& p = pick_a_random_person();
+    const auto context_features = p.get_features();
+    const auto action_features = get_action_features();
+    const auto context_json = create_context_json(context_features,action_features);
+    const auto req_id = create_uuid();  
+    r::api_status status;
+
+    // Choose an action
+    RETURN_ON_ERROR(_rl->choose_rank(req_id.c_str(), context_json.c_str(), response, &status), status);
+    size_t choosen_action;
+    RETURN_ON_ERROR(response.get_choosen_action_id(choosen_action), status);
+
+    // What reward did this action get?
+    const auto reward = p.get_reward(_actions[choosen_action]);
+
+    // Report reward recieved
+    RETURN_ON_ERROR(_rl->report_outcome(req_id.c_str(), reward, &status), status);
+
+    std::cout << "Round:" << round << ", Person:" << p.id() << ", Action:" << choosen_action << ", Reward:" << reward << std::endl;
+    response.clear();
+    ++round;
+  }
+}
+
+person& rl_sim::pick_a_random_person() {
+  return _people[rand() % _people.size()];
+}
+
+int rl_sim::load_config_from_json(  const std::string& file_name, 
```

Comment:
Want to start a discussion on alignment. I feel like it is more consistent and readable to simply move parameters to new lines at a further indent level than aligning like this. But I have no idea how it's been done up until now/if there is convention around this 😄

Reply:
I have seen it used a couple of different ways in this code base.  There does seem to be a tendency towards avoiding extra lines.  (To fit more lines in a small laptop screen perhaps :)